src/Guide/Markdown.hs

```diff
@@ -366,3 +366,107 @@ instance SafeCopy MarkdownTree where
 -- | Is a piece of Markdown empty?
 markdownNull :: HasMdText a Text => a -> Bool
 markdownNull = T.null . view mdText
+
+------------------------------
+------ Markdown Tables -------
+------------------------------
+
+{-|
+Data Structure to hold tables
+-}
+data Table = Table
+           { name    :: Text -- ^ Table Header
+           , columns :: [Text] -- ^ Names of Columns
+           , rows    :: [[Text]] -- ^ List of rows with cells
+           } deriving Show
+
+{-|
+Tries to make 'Table' structure from Node.
+Next markdown
+@
++ %TABLE TableName
++ - Column 1
+  - Column 2
+  - Column 3
++ --------------------------------
+
++ - Foo
+  - Bar
+  - Baz
+
++ - Another foo | Another bar | Another baz
+@
+should be parsed as
+@
+Table
+    { name = "TableName"
+    , columns = ["Column 1", "Column 2", "Column 3"]
+    , rows = [ ["Foo", "Bar", "Baz"]
+             , ["Another Foo", "Another Bar", "Another Baz"]
+             ]
+    }
+@
+-}
+getTable :: MD.Node -> Maybe Table
+getTable (MD.ItemList_ _ (table:columns:brk:rest)) = do
+  let tblName = getTableName table
+  let colNames = getRow columns
+  if getBreak brk then
+    let cellValues = mapM getRow rest in
+    liftA3 Table tblName colNames cellValues
+  else
+    Nothing
+getTable _ = Nothing
+
+{-|
+Parses table name after keyword "%TABLE"
+-}
+getTableName :: [MD.Node] -> Maybe Text
+getTableName [MD.Paragraph_ [MD.Text_ t]] = T.stripPrefix "%TABLE " t
+getTableName _ = Nothing
+
+{-|
+Gets whole row values
+-}
+getRow :: [MD.Node] -> Maybe [Text]
+getRow [MD.ItemList_ _ items] = concat `fmap` mapM getCellFromItem items
+getRow _ = Nothing
+
+{-|
+Possible row syntax  is
+@
++  - smth 1
+   - smth 2
+   - smth 3
+@
+or
+@
+smth 1 | smth 2 | smth 3
+@
+These two examples are equal.
+-}
+getCellFromItem :: [MD.Node] -> Maybe [Text]
```

Comment:
* <code>getCell<b>s</b>FromItem</code>, since there are many returned cells. Maybe just `getCells`, actually.

* A cell should be `[MD.Node]` and not `Text`, since this way users will be able to use markup in table cells. (Breaking on `|` becomes a bit more effort this way, but still fairly easy to do. `*foo* | **bar** | baz` should parse as three cells – [*foo*, **bar**, baz].)

* Also you can't assume that there'll be a paragraph in the list item – if the user tries to add a list or a code block in a table cell, there won't be a paragraph. I recommend http://spec.commonmark.org/dingus/ to quickly check how a piece of Markdown is parsed. If you take this:

   ````
   - ```
     code example
     in a table
     ```
   - some text
   ````

  you will get this AST, without a paragraph in the first item:

  ```xml
  <list type="bullet" tight="true">
    <item>
      <code_block>code example
  in a table
  </code_block>
    </item>
    <item>
      <paragraph>
        <text>some text</text>
      </paragraph>
    </item>
  </list>
   ```

* Finally, this function doesn't even handle lists currently, only the `|` syntax (despite what its documentation suggests).